packages/dm-core-plugins/src/generic-list/GenericListPlugin.tsx

```diff
@@ -161,98 +190,153 @@ export const GenericListPlugin = (
                         }
                       />
-                    </>
-                  )}
-                </td>
-              </TableRow>
-              <TableRow style={{ cursor: 'initial' }}>
-                <td
-                  colSpan={4}
-                  style={{
-                    borderBottom: '1px solid rgba(224, 224, 224, 1)',
-                    textAlign: 'initial',
-                  }}
-                >
-                  {itemsExpanded[key] && (
-                    // @ts-ignore
-                    <ViewCreator
-                      idReference={`${idReference}[${index}]`}
-                      blueprintAttribute={{
-                        name: 'nil',
-                        attributeType: type,
-                        dimensions: '',
-                      }}
-                      viewConfig={
-                        internalConfig.views[index] ??
-                        internalConfig.defaultView
-                      }
-                    />
-                  )}
-                </td>
-              </TableRow>
-            </React.Fragment>
-          ))}
+                    )}
+                  </td>
+                </TableRow>
+              </React.Fragment>
+            )
+          )}
         </tbody>
       </Table>
-      <div>
-        <SaveButton
-          onClick={save}
-          disabled={!unsavedChanges}
-          isLoading={isLoading}
-        />
-        <AppendButton onClick={() => addItem()} />
+
+      <div
+        style={{
+          display: 'grid',
+          gridAutoColumns: 'minmax(0, 1fr)',
+          width: '100%',
+          paddingTop: '10px',
+        }}
+      >
+        <div
+          style={{
+            display: 'flex',
+            flexDirection: 'row',
+            alignItems: 'center',
+          }}
+        >
+          <Label label={'Items per page:'} />
+          <ItemsPerPageButton
+            onClick={setItemsPerPage}
+            setValue={5}
+            current={itemsPerPage}
+          />
+          <ItemsPerPageButton
+            onClick={setItemsPerPage}
+            setValue={10}
+            current={itemsPerPage}
+          />
+          <ItemsPerPageButton
+            onClick={setItemsPerPage}
+            setValue={20}
+            current={itemsPerPage}
+          />
+          <ItemsPerPageButton
+            onClick={setItemsPerPage}
+            setValue={50}
+            current={itemsPerPage}
+          />
+        </div>
+        <div style={{ justifySelf: 'center' }}>
+          <Pagination
+            totalItems={Object.keys(items).length}
+            itemsPerPage={itemsPerPage}
+            defaultPage={currentPage + 1}
+            onChange={(_: any, p: number) => setCurrentPage(p - 1)}
+          />
+        </div>
+        <div
+          style={{
+            display: 'flex',
+            flexDirection: 'row',
+            gridColumnStart: '3',
+            justifySelf: 'end',
+          }}
+        >
```

Comment:
Could you extract this into a separate component in its own file?

Reply:
Replaced with @awesthouse suggestion

Reply:
Going to be a bit harder to get all the stuff aligned on one line. Also no that crucial  now that the "ItemsPerPage" buttons got more compact.